Add ChaosCenter developer guide docs locally (#282)
litmuschaos/litmus-docs: website/docs/developer-guide/chaoscenter-developer-guide.md

---
id: chaoscenter-developer-guide
title: ChaosCenter Developer Guide
sidebar_label: ChaosCenter Developer Guide 
---

---

## **Prerequisites**
:::note

> This document is ietented to be implemented locally. Please do not use in dev or prod environmentss

Reviewer: ```suggestion
This document is intended to be implemented locally. Please do not use it in dev or prod environments.
```

:::

- Kubernetes 1.17 or later
- Helm3 or Kubectl
- Node and npm
- Docker
- Golang
- Local Kubernetes Cluster (via minikube, k3s or kind)

## **Control Plane**
Backend components consist of three microservices

> 1. GraphQL-Server

Reviewer: ```suggestion
1. Backend server
```

Reviewer 2: I think it's fine to have graphql server since we have 2 backend servers

Author: Shoud we use graphql server for the name?? 
in [docs](https://docs.litmuschaos.io/docs/architecture/architecture-summary) using graphql

> 2. Authentication-Server

Reviewer: ```suggestion
2. Authentication server
```

3. MongoDB

Frontend component
1. React

## **Steps to run the Control Plane**

### 1. Run MongoDB

Step-1: Pull and run the image 

```bash
docker pull mongo:5

docker network create mongo-cluster 

docker run -d --net mongo-cluster -p 27015:27015 --name m1 mongo:4.2 mongod --replSet rs0 --port 27015 
docker run -d --net mongo-cluster -p 27016:27016 --name m2 mongo:4.2 mongod --replSet rs0 --port 27016
docker run -d --net mongo-cluster -p 27017:27017 --name m3 mongo:4.2 mongod --replSet rs0 --port 27017
```

Step-2: Add hosts 

Windows
```bash
# add hosts in hosts 
notepad C:\Windows\System32\drivers\etc\hosts

# add the below line
127.0.0.1       m1 m2 m3
```

Linux/Mac
```bash
# add hosts in hosts 
sudo vim /etc/hosts

# add the below line
127.0.0.1       m1 m2 m3
```

Step-3: Configure the mongoDB replica set

```bash
docker exec -it m1 mongo -port 27015

config={"_id":"rs0","members":[{"_id":0,"host":"m1:27015"},{"_id":1,"host":"m2:27016"},{"_id":2,"host":"m3:27017"}]}

rs.initiate(config)

db.getSiblingDB("admin").createUser({user:"admin",pwd:"1234",roles:[{role:"root",db:"admin"}]});

> ```

Reviewer: We should use Docusaurus tabs for individual infra types to show them. Please refactor in all the places.

export DB_USER=admin
export DB_PASSWORD=1234
export JWT_SECRET=litmus-portal@123
export PORTAL_ENDPOINT=http://localhost:8080
export LITMUS_SVC_ENDPOINT=""
export SELF_AGENT=false
export INFRA_SCOPE=cluster
export INFRA_NAMESPACE=litmus
export LITMUS_PORTAL_NAMESPACE=litmus
export PORTAL_SCOPE=namespace
export SUBSCRIBER_IMAGE=litmuschaos/litmusportal-subscriber:ci
export EVENT_TRACKER_IMAGE=litmuschaos/litmusportal-event-tracker:ci
export CONTAINER_RUNTIME_EXECUTOR=k8sapi
export ARGO_WORKFLOW_CONTROLLER_IMAGE=argoproj/workflow-controller:v2.11.0
export ARGO_WORKFLOW_EXECUTOR_IMAGE=argoproj/argoexec:v2.11.0
export CHAOS_CENTER_SCOPE=cluster
export WORKFLOW_HELPER_IMAGE_VERSION=3.0.0
export LITMUS_CHAOS_OPERATOR_IMAGE=litmuschaos/chaos-operator:3.0.0
export LITMUS_CHAOS_RUNNER_IMAGE=litmuschaos/chaos-runner:3.0.0
export LITMUS_CHAOS_EXPORTER_IMAGE=litmuschaos/chaos-exporter:3.0.0
export ADMIN_USERNAME=admin
export ADMIN_PASSWORD=litmus
export VERSION=ci
export HUB_BRANCH_NAME=v2.0.x
export INFRA_DEPLOYMENTS="[\"app=chaos-exporter\", \"name=chaos-operator\", \"app=event-tracker\",\"app=workflow-controller\"]"
export INFRA_COMPATIBLE_VERSIONS='["0.2.0", "0.1.0","ci"]'

> export DEFAULT_HUB_BRANCH_NAME=master

Reviewer 2: We have updated the envs in the wiki, can you update them here as well?

Author: yes, sure. I'll update
